zip-0244.rst

```diff
@@ -435,37 +548,32 @@ If the sighash type is ``SIGHASH_SINGLE`` and the signature hash is being comput
 the transparent input at a particular index, and a transparent output appears in
 the transaction at that index::
 
-   the hash is updated with the transaction serialized form of the
-   transparent output at that index, and finalized.
+   the hash is over the transaction serialized form of the transparent output at that
+   index
 
-If the sighash type is ``SIGHASH_SINGLE`` and the signature is being computed for
-a shielded input, or if the sighash type is ``SIGHASH_NONE``::
+otherwise::
 
-   the hash is immediately finalized, without being updated with any
-   additional data
+   BLAKE2b-256(``ZTxIdOutputsHash``, [])
 
 S.2d: txin_sig_digest
 '''''''''''''''''''''
-This is a BLAKE2b-256 hash initialized with the personalization field value
-``Zcash___TxInHash`` (3 underscores).
-
-If the signature hash is being computed for a transparent input, the hash
-is updated with the following properties of that input::
+This is a BLAKE2b-256 hash of the following properties of the transparent input being
+signed, initialized with the personalization field value ``Zcash___TxInHash`` (3
+underscores).
 
    S.2d.i:   prevout     (field encoding)
    S.2d.ii:  script_code (field encoding)
    S.2d.iii: value       (8-byte signed little-endian)
    S.2d.iv:  nSequence   (4-byte unsigned little-endian)
 
-otherwise::
-
-   the hash is immediately finalized, without being updated with any
-   additional data
-
```

Comment:
Should there be an "otherwise: empty hash" here as well?

Reply:
It is not necessary, because if the input being signed is not a transparent input, S.2 specifies that the hash be identical to T.2

Reply:
Yep, so this sub-function will never be called except when there is a transparent input that requires digesting.